addons/stock_account/models/account_move.py

```diff
@@ -252,8 +252,8 @@ def _eligible_for_cogs(self):
         return self.product_id.type == 'product' and self.product_id.valuation == 'real_time'
 
     def _get_gross_unit_price(self):
-        price_unit = -self.price_unit if self.move_id.move_type == 'in_refund' else self.price_unit
-        price_unit = price_unit * (1 - (self.discount or 0.0) / 100.0)
+        price_unit = self.price_subtotal / self.quantity
```

Comment:
@amoyaux are we absolutely sure quantity can never be zero?

Reply:
I tried a bit, and it's seems that either the stock.move is not created or cancel in that case